src/dmsort.rs

```diff
@@ -266,7 +267,9 @@ where
 				|| compare(s.slice.get_unchecked(read), s.slice.get_unchecked(s.write - 1)) != Ordering::Less
 			{
 				// The element is order - keep it:
-				unsafe_copy(s.slice, read, s.write);
+				if read != s.write {
+					unsafe_copy(s.slice, read, s.write);
+				}
```

Comment:
Some of the calls here to `unsafe_copy` pass `0` for both the read and write index, making this an overlapping copy.